pkg/controllers/allocation/scheduling/preferences.go

```diff
@@ -0,0 +1,92 @@
+package scheduling
+
+import (
+	"context"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	ExpirationTTL   = 5 * time.Minute
+	CleanupInterval = 1 * time.Minute
+)
+
+type Preferences struct {
+	cache *cache.Cache
+}
+
+func NewPreferences() *Preferences {
+	return &Preferences{
+		cache: cache.New(ExpirationTTL, CleanupInterval),
+	}
+}
+
+// Relax removes soft preferences from pods to enable scheduling if the cloud
+// provider's capacity is constrained. For example, this can be leveraged to
+// prefer a specific zone, but relax the preferences if the pod cannot be
+// scheduled to that zone. Preferences are removed iteratively until only hard
+// constraints remain. Pods relaxation is reset (forgotten) after 5 minutes.
```

Comment:
5 minutes is just the default I think? (Comment makes it seem hard-coded)

Reply:
Currently is hard coded. I'm not sure we want to expose this to the customer, as I can't imagine how they'd decide to set it.